File: color-spaces/RGBtoHSV/RGBtoHSV.py
# -----------------------------------------------------------------------------
#
# Example for translation image from RGB to HSV color space on Python3 with OpenCV.
#
# -----------------------------------------------------------------------------

import cv2

imageRGB = cv2.imread("image.png");

if imageRGB is None:
    exit(-1)

imageHSV = cv2.cvtColor(imageRGB, cv2.COLOR_RGB2HSV);

cv2.imwrite("imageHSV.png", imageHSV);

File: color-spaces/RGBtoHSV/rgb_to_hsv.cpp
//-----------------------------------------------------------------------------
//
// Example for translation image from RGB to HSV color space on C++ with OpenCV.
//
//-----------------------------------------------------------------------------

#include <opencv2/opencv.hpp>

int main() {
    cv::Mat imageRGB = cv::imread("image.png");

    if (imageRGB.empty()) {
        std::cerr << "Error: Could not read the image file" << std::endl;
        return -1;
    }

    cv::Mat imageHSV;
    cv::cvtColor(imageRGB, imageHSV, cv::COLOR_RGB2HSV);

    cv::imwrite("imageHSV.png", imageHSV);
}
